lib/Onyx.js

```diff
@@ -1025,17 +1028,22 @@ function initializeWithDefaultKeyStates() {
  * Storage.setItem() from Onyx.clear() will have already finished and the merged
  * value will be saved to storage after the default value.
  *
+ * @param {Array} keysToPreserve ONYXKEYS which values you want to keep
  * @returns {Promise<void>}
  */
-function clear() {
+function clear(keysToPreserve = []) {
     return getAllKeys()
         .then((keys) => {
-            _.each(keys, (key) => {
-                const resetValue = lodashGet(defaultKeyStates, key, null);
-                cache.set(key, resetValue);
-                notifySubscribersOnNextTick(key, resetValue);
+            const defaultKeys = _.keys(defaultKeyStates);
+            const keysToClear = _.without(keys, ..._.union(keysToPreserve, defaultKeys));
```

Comment:
The [without](https://underscorejs.org/#without) function takes in a variable amount of arguments to omit. If we didn't use the spread operator it would attempt finding that _entire_ array of `_.union(keysToPreserve, defaultKeys)` and removing that from `keys`.

Reply:
OK, I see. I think it might make more sense to do this since it removes both the union and the spread:
```suggestion
            const keysToClear = _.difference(keys, keysToPreserve, defaultKeys);
```